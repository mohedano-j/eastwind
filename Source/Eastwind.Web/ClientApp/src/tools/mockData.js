const products= [

  {
    productId: 1,
    productName: "test1",
    categoryId: 1,
    unitPrice: 12,
    unitsInStock:  11,
    categoryName: "Food"
  },
  {
    productId: 2,
    productName: "test2",
    categoryId: 2,
    unitPrice: 10,
    unitsInStock:  10,
    categoryName: "Drinks"
  }

];

const categories = [{
  categoryId: 1,
  categoryName: "Food",
  description: "Food Description",
  picture: []
},{
  categoryId: 2,
  categoryName: "Drinks",
  description: "Drinks description",
  picture: []
}
];

const newProduct = {
  productId: 0,
  productName: "",
  categoryId: 0,
  unitPrice: 0,
  unitsInStock:  0,
  categoryName: ""
};

// Using CommonJS style export so we can consume via Node (without using Babel-node)
module.exports = {
  newProduct,
  products,
  categories
};
