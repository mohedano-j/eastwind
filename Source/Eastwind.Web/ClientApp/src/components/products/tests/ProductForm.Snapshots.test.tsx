import React from "react";
import ProductForm from "../ProductForm";
import renderer from "react-test-renderer";
import { products,categories } from "../../../tools/mockData";

it("sets submit button label 'Saving...' when saving is true", () => {
  const tree = renderer.create(
    <ProductForm
      product={products[0]}
      categories={categories}
      onSave={jest.fn()}
      onChange={jest.fn()}
      onCancel={jest.fn()}
      errors={[]}
      saving
    />
  );

  expect(tree).toMatchSnapshot();
});

it("sets submit button label 'Save' when saving is false", () => {
  const tree = renderer.create(
    <ProductForm
      product={products[0]}
      categories={categories}
      onSave={jest.fn()}
      onChange={jest.fn()}
      onCancel={jest.fn()}
      errors={[]}
      saving={false}
    />
  );

  expect(tree).toMatchSnapshot();
});