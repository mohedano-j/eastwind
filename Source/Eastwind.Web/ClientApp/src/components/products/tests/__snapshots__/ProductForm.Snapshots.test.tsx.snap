// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sets submit button label 'Save' when saving is false 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <h2>
    Edit
     Product
  </h2>
  <div
    className="form-group"
  >
    <label
      htmlFor="productName"
    >
      Name
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="productName"
        onChange={[MockFunction]}
        type="text"
        value="test1"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="categoryId"
    >
      Category
    </label>
    <div
      className="field"
    >
      <select
        className="form-control"
        name="categoryId"
        onChange={[MockFunction]}
        value={1}
      >
        <option
          value=""
        >
          Select Category
        </option>
        <option
          value={1}
        >
          Food
        </option>
        <option
          value={2}
        >
          Drinks
        </option>
      </select>
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="unitPrice"
    >
      Unit Price
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="unitPrice"
        onChange={[MockFunction]}
        type="text"
        value={12}
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="unitsInStock"
    >
      Units In Stocks
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="unitsInStock"
        onChange={[MockFunction]}
        type="text"
        value={11}
      />
    </div>
  </div>
  <div>
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit"
    >
      Save
    </button>
    <button
      className="btn btn-light ml-2"
      onClick={[MockFunction]}
    >
      Cancel
    </button>
  </div>
</form>
`;

exports[`sets submit button label 'Saving...' when saving is true 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <h2>
    Edit
     Product
  </h2>
  <div
    className="form-group"
  >
    <label
      htmlFor="productName"
    >
      Name
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="productName"
        onChange={[MockFunction]}
        type="text"
        value="test1"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="categoryId"
    >
      Category
    </label>
    <div
      className="field"
    >
      <select
        className="form-control"
        name="categoryId"
        onChange={[MockFunction]}
        value={1}
      >
        <option
          value=""
        >
          Select Category
        </option>
        <option
          value={1}
        >
          Food
        </option>
        <option
          value={2}
        >
          Drinks
        </option>
      </select>
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="unitPrice"
    >
      Unit Price
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="unitPrice"
        onChange={[MockFunction]}
        type="text"
        value={12}
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="unitsInStock"
    >
      Units In Stocks
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="unitsInStock"
        onChange={[MockFunction]}
        type="text"
        value={11}
      />
    </div>
  </div>
  <div>
    <button
      className="btn btn-primary"
      disabled={true}
      type="submit"
    >
      Saving...
    </button>
    <button
      className="btn btn-light ml-2"
      onClick={[MockFunction]}
    >
      Cancel
    </button>
  </div>
</form>
`;
